src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -861,18 +844,15 @@ void CastSpellInfoHelpers::castSpell() {
                     if (spell_targeted_at &&
                             PID_TYPE(spell_targeted_at) == OBJECT_Actor) {
                         int monster_id = PID_ID(spell_targeted_at);
-                        dist_X = abs(pActors[monster_id].vPosition.x - pParty->vPosition.x);
-                        dist_Y = abs(pActors[monster_id].vPosition.y - pParty->vPosition.y);
-                        dist_Z = abs(pActors[monster_id].vPosition.z - pParty->vPosition.z);
-                        int count = int_get_vector_length(dist_X, dist_Y, dist_Z);
+                        Vec3i dist = (pActors[monster_id].vPosition - pParty->vPosition).Abs();
```

Comment:
float dist = (pActors[monster_id].vPosition - pParty->vPosition).ToFloat().Length();

And don't add the `Abs` method.

Then just compare dist with `307.2` below.